src/read-only-handler.ts

```diff
@@ -1,37 +1,38 @@
-import {
-    isUndefined,
-    ArrayConcat,
-    ObjectDefineProperty,
-    getOwnPropertyDescriptor,
-    getOwnPropertyNames,
-    getOwnPropertySymbols,
-    hasOwnProperty,
-} from './shared';
+import { unwrap, freeze } from './shared';
+import { BaseProxyHandler, ReactiveMembraneShadowTarget } from './base-handler';
 
-import {
-    ReactiveMembrane,
-    ReactiveMembraneShadowTarget,
-    wrapDescriptor,
-} from './reactive-membrane';
+const getterMap = new WeakMap<() => any, () => any>();
+const setterMap = new WeakMap<(v: any) => void, (v: any) => void>();
 
-function wrapReadOnlyValue(membrane: ReactiveMembrane, value: any): any {
-    return membrane.valueIsObservable(value) ? membrane.getReadOnlyProxy(value) : value;
-}
-
-export class ReadOnlyHandler {
-    private originalTarget: any;
-    private membrane: ReactiveMembrane;
-
-    constructor(membrane: ReactiveMembrane, value: any) {
-        this.originalTarget = value;
-        this.membrane = membrane;
+export class ReadOnlyHandler extends BaseProxyHandler {
+    wrapValue(value: any): any {
+        return this.membrane.getReadOnlyProxy(value);
     }
-    get(shadowTarget: ReactiveMembraneShadowTarget, key: PropertyKey): any {
-        const { membrane, originalTarget } = this;
-        const value = originalTarget[key];
-        const { valueObserved } = membrane;
-        valueObserved(originalTarget, key);
-        return membrane.getReadOnlyProxy(value);
+    wrapGetter(originalGet: () => any): () => any {
```

Comment:
this function is identical to that of reactive-handler.ts, perhaps this function can move to BaseProxyHandler class or a utility?